azkaban-common/src/main/java/azkaban/imagemgmt/daos/RampRuleDao.java

```diff
@@ -42,13 +43,25 @@ public interface RampRuleDao {
    */
   int addRampRule(final ImageRampRule rule);
 
+  /**
+   * Update owners metadata into table ramp_rules.
+   *
+   * @param newOwners
+   * @param ruleName
+   * @param modifiedBy
+   * @return int - id of the DB entry
+   * @throws azkaban.imagemgmt.exception.ImageMgmtDaoException
+   */
+  int updateOwnerships(final String newOwners, final String ruleName, final String modifiedBy);
+
   /**
    * Query table ramp_rules to get owners of Ramp rule.
+   * Serves as the source of truth for rule ownership management.
    *
    * @param ruleName - ruleName in {@see ImageRampRule}
    * @return owners of the ramp rule.
    */
-  String getOwners(final String ruleName);
+  Set<String> getOwners(final String ruleName);
```

Comment:
Minor: since these changes also included in PR, maybe also mention them in title, or consider the next time have 2 separate commits / PRs

Reply:
long story short: 
the reason I added ownershipManagement part into this PR is now the ownership management is different than the original design.
In the previous design doc, there is a plan to maintaining an independent hp_flow_ownership table just as image_ownerships. I abandoned this for two reasons:
- hard to fetch ownership during HP rule creation as lack of flow information;
- introducing high priority flow service into the system is not necessary, we only want hp_owners so let's just keep this ownership within rampRule to make the ownership more straightforward and easy managable.

So in order to address the difference to change the ownership source of truth from maintaining a table to a column in ramp rules. I added support to modify ownerships too, comparing to the previous design that all the ownership would need to fetch from hp_flow_ownership and image_ownership tables.